ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -8292,8 +8294,8 @@ private WriteEntity generateTableWriteEntity(String dest, Table dest_tab,
     if ((dpCtx == null || dpCtx.getNumDPCols() == 0)) {
       output = new WriteEntity(dest_tab, determineWriteType(ltd, dest));
       if (!outputs.add(output)) {
-        if(!((this instanceof MergeSemanticAnalyzer) &&
-            conf.getBoolVar(ConfVars.MERGE_SPLIT_UPDATE))) {
+        if(!((this instanceof MergeSemanticAnalyzer || this instanceof UpdateDeleteSemanticAnalyzer) &&
```

Comment:
I believe this could be more readable if the condition would be placed into an instance method; which could be overriden by `MergeSemanticAnalyzer` / `UpdateDeleteSemanticAnalyzer`

Reply:
Extracted to instance method.